Solutions/Digital Shadows/Data Connectors/Digital Shadows/DigitalShadowsConnectorAzureFunction/DS_poller.py

```diff
@@ -0,0 +1,155 @@
+        arr = data.splitlines()
+        res = ""
+        for e in arr:
+            res = res + e + "\n\n"
+        return res
+
+
+    def post_azure(self, response, item):
+        """
+            posts to azure after appending triage information on it
+        """
+        json_obj = response.json()
+        for it in item:
+            for i in range(len(json_obj)):
+                if(it['source']['alert-id'] == json_obj[i]['id'] or it['source']['incident-id'] == json_obj[i]['id']):
+
+                    comment_data = self.DS_obj.get_triage_comments(it['id'])
+                    json_obj[i]['status'] = it['state']
+                    json_obj[i]['triage_id'] = it['id']
+                    json_obj[i]['triage_raised_time'] = it['raised']
+                    json_obj[i]['triage_updated_time'] = it['updated']
+                    
+                    comment_data_filtered = []
+                    for comment in comment_data:
+                        comment['user-name'] = comment['user']['name']
+                        del comment['triage-item-id']
+                        del comment['updated']
+                        del comment['user']
+                        if comment['content'] != "":
+                            comment_data_filtered.append(comment)
+
+
+                    json_obj[i]['comments'] = comment_data_filtered
+                    
+                    json_obj[i]['description'] = self.parse_desc(json_obj[i]['description'])
+
+                    if('id' in json_obj[i] and not isinstance(json_obj[i]['id'], str)):
+                        json_obj[i]['description'] = json_obj[i]['description'] + "\n\nSearchlight Portal Link: https://portal-digitalshadows.com/triage/alert-incidents/" + str(json_obj[i]['id'])
+                
+                    self.AS_obj.post_data(json.dumps(json_obj[i]), constant.LOG_NAME)
+
+    def get_data(self):
+        """
+            getting the incident and alert data from digital shadows
+        """
+        triage_id = []
+
+        try:
+            if(isinstance(self.event, int)):
+                event_data = self.DS_obj.get_triage_events_by_num(self.event)
+                #calculating the max event number from current batch to  use in next call
+                if(event_data):
+                    max_event_num = max([e['event-num'] for e in event_data])
+
+            else:
+                event_data = self.DS_obj.get_triage_events(self.before_time, self.after_time)
+                #calculating the max event number from current batch to  use in next call
+                if(event_data):
+                    max_event_num = max([e['event-num'] for e in event_data])
+                    logger.info("First poll from event number " + str(event_data[0]['event-num']))
+                    logger.info("Total number of events are " + str(len(event_data)))
+                else:
+                    logger.info("There are no new events")
+                    return [], int
```

Comment:
shouldn't return the int object here.

Perhaps initialise `max_event_num` to `-1` at the start of the method and if then we don't need this else statement.

In the calling method we can look at `max_event_num` and if it is `-1` we know there's nothing to look at. Although I can see you are already checking to see if there are any items in `event_data`

We probably just remove this `else` block - you don't have a matching on in the `if` block above and don't need it here..